### PR TITLE
Don't delete local socket when reverse

### LINKED ISSUE #1724: Reverse unix socket forwarding in Mac deletes the source unix socket.
### Description

limactl version 0.16.0

I'm trying to run Docker Desktop on my M2 Mac, and connecting to its unix socket from inside the lima vm. So I created this configuration:

```
portForwards:
- guestSocket: "/run/user/{{.UID}}/docker.sock"
  hostSocket: "/Users/rangelreale/.docker/run/docker.sock"
  reverse: true
```

it seems to work, but the unix socket stops work both in the host and in the guest
host:
```
$ docker container ls
Cannot connect to the Docker daemon at unix:///Users/rangelreale/.docker/run/docker.sock. Is the docker daemon running?
```

guest:
```
docker container ls
error during connect: Get "http://%2Frun%2Fuser%2F501%2Fdocker.sock/v1.24/containers/json": read unix @->/run/user/501/docker.sock: read: connection reset by peer
```

It seems that when using `reverse: true`, the host socket is being deleted, according to @afbjorklund in slack.

### PR DESCRIPTION
**Pull request by afbjorklund**
When we are forwarding a local socket, we should not delete it.

Theoretically we could delete the remote socket, if it exists.

Closes #1724

* https://github.com/lima-vm/lima/pull/836

### CODE REVIEW

**Comment by afbjorklund** [comment]
Actually we _should_ delete the remote socket, otherwise restart fails.

WARN[0031] [hostagent] Failed to set up forward from "/var/tmp/docker.sock" (guest) to "/var/run/docker.sock" (host)

File: pkg/hostagent/hostagent.go [hunk 1]
```diff
@@ -568,6 +568,21 @@ const (
...
+		"127.0.0.1",
+		"--",
+	)
+	args = append(args, command...)
+	cmd := exec.CommandContext(ctx, sshConfig.Binary(), args...)
+	if out, err := cmd.Output(); err != nil {
+		return fmt.Errorf("failed to run %v: %q: %w", cmd.Args, string(out), err)
+	}
+	return nil
+}
```
Comment (AkihiroSuda) [inline pkg/hostagent/hostagent.go:584]: Shouldn't we use `sshutil.SSHOpts` ?

Comment (afbjorklund) [reply pkg/hostagent/hostagent.go:584]: We are, it is getting passed in sshConfig as far as I know?

```go
        sshConfig := &ssh.SSHConfig{
                AdditionalArgs: sshutil.SSHArgsFromOpts(sshOpts),
        }
```

**Review by AkihiroSuda — APPROVED** [review]
Thankss

**Comment by AkihiroSuda** [comment]
Rebasing may fix the CI failure

**Review by AkihiroSuda — APPROVED** [review]
Thanks, but we should have a test for this

**Pull request merged**